### PR TITLE
v1

### PR DESCRIPTION
**Pull request by andreigec**
v1

### CODE REVIEW

File: src/components/ErrorBoundary/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React           from 'react';
+import PropTypes       from 'prop-types';
+import { error as eF } from '../../helpers';
+
+class ErrorBoundary extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false };
```
Comment (tripleaxis) [inline src/components/ErrorBoundary/index.js]: Would be simpler to just have a `state` class prop

File: src/components/ErrorBoundary/index.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React           from 'react';
+import PropTypes       from 'prop-types';
+import { error as eF } from '../../helpers';
```
Comment (tripleaxis) [inline src/components/ErrorBoundary/index.js]: `eF` is a meaningless name - better to use something to describe what it does

File: src/components/ErrorPage/errorpage.style.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { styled } from '../../styles';
+import * as F     from '../../styles/fonts';
```
Comment (tripleaxis) [inline src/components/ErrorPage/errorpage.style.js]: `F` and `C` - as above, pls use descriptive names.

File: src/components/ErrorPage/index.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+import { SBase }       from './errorpage.style';
+
+const errorPage = () => (
+  <SBase>
+    Unfortunately an error has occured (500). This error has been sent to the administrator, and
+    will be fixed shortly!
+  </SBase>
+);
+
+export default memo(errorPage);
```
Comment (tripleaxis) [inline src/components/ErrorPage/index.js]: Pretty sure `memo` is useless for this component - no props, so wouldn’t re-render anyway.

File: src/components/Link/Link.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React        from 'react';
+import { addRoute } from './store';
+
+const Link = (pr) => (
+  <a
```
Comment (tripleaxis) [inline src/components/Link/Link.js]: Pls refactor this for readability. Very difficult to see what’s happening here - difficult to maintain.

Also, what is `pr`? Props?
Consider clearer param name

File: src/components/Link/Link.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    {...pr}
+    onClick={(e) => {
```
Comment (tripleaxis) [inline src/components/Link/Link.js]: Not obvious that `pr.onClick` is overridden by this static `onClick` - consider destructuring your param to remove any props you need to control.

`const { onClick, ...linkProps } = pr`

File: src/components/Link/linkHooks.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  const origstate = {
+    userAgent,
+    pathName,
+    search,
+    myhistory: [],
+    routes,
+    route: getRoute(routes, pathName),
+  };
+  const [data, setData] = useState(origstate);
+  const ue = (locationOverride) => {
```
Comment (tripleaxis) [inline src/components/Link/linkHooks.js]: Looks like this is intended to emulate `componentDidMount`. Would be best to either name it appropriately or add a comment explaining its purpose.

File: src/components/Link/linkHooks.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
+import { toQuery, parseQuery }                                from '../../helpers';
+import {
+  addRoute, getRoute, historyListen, historyUnListen,
+} from './store';
+
+export default function linkHooks(pr) {
```
Comment (tripleaxis) [inline src/components/Link/linkHooks.js]: This function needs some cleaning up for readability/maintainability. Suggest to add some formatting/line breaks to add clarity. 

May want to add prettier or similar to help with this.

File: src/components/WaitingAnim/index.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+  componentDidMount() {
+    this.timer = setTimeout(() => this.setState({ transparent: true }), 100);
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timer);
+  }
+
+  render() {
```
Comment (tripleaxis) [inline src/components/WaitingAnim/index.js]: Nice 👍. cleanly written and good use of short-circuit returns to reduce complexity.

File: src/containers/App/index.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        <Helmet>
+          {[
+            <meta charSet="utf-8" key="h1" />,
+            <meta name="description" content="SKELETON" key="h2" />,
+            <title key="h3">SKELETON</title>,
+          ]}
+        </Helmet>
+      </HelmetProvider>
+
+      <S.SMain id="main">
```
Comment (tripleaxis) [inline src/containers/App/index.js]: `S.SMain` Dude - naming needs more consideration 😉

File: src/containers/App/index.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        <>
```
Comment (tripleaxis) [inline src/containers/App/index.js]: Why is this fragment here? Doesn’t look necessary

File: package.json [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    "stylelint-config-andreigec": "^1.0.5",
+    "stylelint-config-recommended": "^2.1.0",
+    "stylelint-config-standard": "^18.2.0",
+    "stylelint-config-styled-components": "^0.1.1",
+    "stylelint-declaration-strict-value": "^1.1.2",
+    "stylelint-processor-styled-components": "^1.5.2",
+    "stylelint-selector-bem-pattern": "^2.0.0"
+  },
+  "author": "andrei gec",
+  "license": "MIT in LICENSE.md",
```
[the remark this answers is not in the document]
Comment (andreigec) [reply package.json]: https://docs.npmjs.com/files/package.json

If you are using a license that hasn’t been assigned an SPDX identifier, or if you are using a custom license, use a string value like this one:

{ "license" : "SEE LICENSE IN <filename>" }

[the remark this answers is not in the document]
Comment (RyanTheAllmighty) [reply package.json]: Yeah but you're using `MIT` so why not just use `MIT`?

File: src/components/WaitingAnim/index.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+  SRect3,
+  SRect4,
+  SRect5,
+  SSpin,
+} from './waitinganim.style';
+
+class WaitingAnim extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
```
[the remark this answers is not in the document]
Comment (andreigec) [reply src/components/WaitingAnim/index.js]: good stuff

File: public/index.html [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+              if (r) {
+                r.forEach(r2 => r2.unregister());
+              }
+            } catch (e) {
+              console.error("error unreg sw:" + e);
+            }
+          }
+        } catch (e) {}
+        window.location.reload(true);
+      }, 5000);
```
[the remark this answers is not in the document]
Comment (andreigec) [reply public/index.html]: whats a reasonable alternative to reload/refresh the page/cache/service workers if a page error occurs?

File: package.json (hunk 12 above)
[the remark this answers is not in the document]
Comment (andreigec) [reply package.json]: fair

**Comment by andreigec** [comment]
> * Variable/import naming needs to be addressed
> * lots of instances where code needs formatting to aid readability and clarity. Recommend adding prettier to the project.
> * general structure seems ok - haven’t checked-out and run the code though.
> * Where are the tests?

**Pull request closed without merging**